src/OrbitGl/TimeGraph.cpp

```diff
@@ -481,6 +485,59 @@ void TimeGraph::ProcessMemoryTrackingTimer(const TimerInfo& timer_info) {
   }
 }
 
+void TimeGraph::ProcessCGroupAndProcessMemoryTrackingTimer(const TimerInfo& timer_info) {
+  int64_t cgroup_limit_bytes = orbit_api::Decode<int64_t>(timer_info.registers(
+      static_cast<size_t>(OrbitApp::CGroupAndProcessMemoryUsageEncodingIndex::kCGroupLimitBytes)));
+  int64_t cgroup_rss_bytes = orbit_api::Decode<int64_t>(timer_info.registers(
+      static_cast<size_t>(OrbitApp::CGroupAndProcessMemoryUsageEncodingIndex::kCGroupRssBytes)));
+  int64_t cgroup_mapped_file_bytes =
+      orbit_api::Decode<int64_t>(timer_info.registers(static_cast<size_t>(
+          OrbitApp::CGroupAndProcessMemoryUsageEncodingIndex::kCGroupMappedFileBytes)));
+  int64_t process_rss_anon_kb = orbit_api::Decode<int64_t>(timer_info.registers(
+      static_cast<size_t>(OrbitApp::CGroupAndProcessMemoryUsageEncodingIndex::kProcessRssAnonKb)));
+
+  if (cgroup_limit_bytes == kMissingInfo || cgroup_rss_bytes == kMissingInfo ||
+      cgroup_mapped_file_bytes == kMissingInfo || process_rss_anon_kb == kMissingInfo) {
+    return;
+  }
+
+  constexpr double kMegabytesToBytes = 1024.0 * 1024.0;
+  constexpr double kMegabytesToKilobytes = 1024.0;
+  orbit_gl::CGroupAndProcessMemoryTrack* track = track_manager_->GetCGroupAndProcessMemoryTrack();
+  if (track == nullptr) {
+    const std::array<std::string, orbit_gl::kCGroupAndProcessMemoryTrackDimension> kSeriesNames = {
+        absl::StrFormat("Process [%s] Resident Anonymous Memory", capture_data_->process_name()),
+        "Other Processes Resident Anonymous Memory",
+        absl::StrFormat("CGroup [%s] Mapped File", timer_info.cgroup_name()),
+        absl::StrFormat("CGroup [%s] Unused", timer_info.cgroup_name())};
+    track = track_manager_->CreateAndGetCGroupAndProcessMemoryTrack(kSeriesNames);
+  }
+  double cgroup_limit_mb = static_cast<double>(cgroup_limit_bytes) / kMegabytesToBytes;
+  double cgroup_rss_anon_mb = static_cast<double>(cgroup_rss_bytes) / kMegabytesToBytes;
+  double cgroup_mapped_file_mb = static_cast<double>(cgroup_mapped_file_bytes) / kMegabytesToBytes;
+  double process_rss_anon_mb = static_cast<double>(process_rss_anon_kb) / kMegabytesToKilobytes;
+  double other_rss_anon_mb = cgroup_rss_anon_mb - process_rss_anon_mb;
+  double unused_mb = cgroup_limit_mb - cgroup_rss_anon_mb - cgroup_mapped_file_mb;
+  track->AddValues(timer_info.start(),
+                   {process_rss_anon_mb, other_rss_anon_mb, cgroup_mapped_file_mb, unused_mb});
+  track->OnTimer(timer_info);
+
+  if (!track->GetValueUpperBound().has_value()) {
```

Comment:
I know this was before in the other track but if it has a value we are not updating it? So, can I assume than this cgroup_limit_bytes is a value who doesn't change?

Reply:
We can update the cgroup limit but we usually don't do this. We can assume that `cgroup_limit_bytes` is a fixed value at least won't change while taking the capture.

Reply:
Ok, so I would like to not have this every time we process a new timer. Instead moving this logic to the constructor of this track, and having a method to set it again. Not to do in this PR, but should be part of the refactor planned from the tracks and the timers. Could you add a TODO?

Reply:
Thanks! I have added a todo and created a bug for this!